app/src/lib/stores/commit-status-store.ts

```diff
@@ -380,6 +417,154 @@ export class CommitStatusStore {
+    repository: GitHubRepository,
+    ref: string,
+    branchName: string,
+    checkRuns: ReadonlyArray<IRefCheck>
+  ): Promise<ReadonlyArray<IRefCheck>> {
+    const key = getCacheKeyForRepository(repository, ref)
+    const subscription = this.subscriptions.get(key)
+    if (subscription === undefined) {
+      return checkRuns
+    }
+
+    const { endpoint, owner, name } = subscription
+    const account = this.accounts.find(a => a.endpoint === endpoint)
+    if (account === undefined) {
+      return checkRuns
+    }
+
+    const api = API.fromAccount(account)
+    const latestWorkflowRuns = await this.getLatestPRWorkflowRuns(
+      api,
+      owner,
+      name,
+      branchName
+    )
+
+    if (latestWorkflowRuns.length === 0) {
+      return checkRuns
+    }
+
+    const logHash = new Map<string, JSZip>()
+    const mappedCheckRuns = new Array<IRefCheck>()
+    for (let i = 0; i < checkRuns.length; i++) {
+      const cr = checkRuns[i]
+      const matchingWR = latestWorkflowRuns.find(
+        wr => wr.check_suite_id === cr.checkSuiteId
+      )
+      if (matchingWR === undefined) {
+        mappedCheckRuns.push(cr)
+        continue
+      }
+
+      const workFlowRunJobs = await api.fetchWorkflowRunJobs(matchingWR)
+      const matchingJob = workFlowRunJobs?.jobs.find(j => j.name === cr.name)
+      if (matchingJob === undefined) {
+        mappedCheckRuns.push(cr)
+        continue
+      }
+
+      // One workflow can have the logs for multiple check runs.. no need to
+      // keep retrieving it. So we are hashing it.
+      const logZip =
+        logHash.get(matchingWR.name) ??
+        (await api.fetchWorkflowRunJobLogs(matchingWR.logs_url))
+      if (logZip === null) {
+        mappedCheckRuns.push(cr)
+        continue
+      }
+
+      logHash.set(matchingWR.name, logZip)
+
+      mappedCheckRuns.push({
+        ...cr,
+        htmlUrl: matchingJob.html_url,
+        output: {
+          ...cr.output,
+          type: RefCheckOutputType.Actions,
+          steps: await parseJobStepLogs(logZip, matchingJob),
+        },
+      })
+    }
+
+    return mappedCheckRuns
+  }
+
+  // Gets only the latest PR workflow runs hashed by name
+  private async getLatestPRWorkflowRuns(
+    api: API,
+    owner: string,
+    name: string,
+    branchName: string
+  ): Promise<ReadonlyArray<IAPIWorkflowRun>> {
+    const wrMap = new Map<string, IAPIWorkflowRun>()
+    const allBranchWorkflowRuns = await api.fetchPRWorkflowRuns(
+      owner,
+      name,
+      branchName
+    )
+
+    if (allBranchWorkflowRuns === null) {
+      return []
+    }
+
+    // When retrieving Actions Workflow runs it returns all present and past
+    // workflow runs for the given branch name. For each workflow name, we only
+    // care about showing the latest run.
+    allBranchWorkflowRuns.workflow_runs.forEach(wr => {
+      const storedWR = wrMap.get(wr.name)
+      if (storedWR === undefined) {
+        wrMap.set(wr.name, wr)
+        return
```

Comment:
If you 👍  my suggestion above, this needs to be changed to `continue`
```suggestion
        continue
```

EDIT: I just noticed I wrote `break` here instead of `continue` as I did in my previous message 🤦